internal/querynodev2/handlers.go

```diff
@@ -457,6 +457,29 @@ func (node *QueryNode) getChannelStatistics(ctx context.Context, req *querypb.Ge
 	return resp, nil
 }
 
+func (node *QueryNode) tryReleaseCollection(collectionID int64) {
+	collection := node.manager.Collection.Get(collectionID)
+	if collection == nil {
+		return
+	}
+
+	delegatorFound := false
+	node.delegators.Range(func(shard string, delegator delegator.ShardDelegator) bool {
+		delegatorFound = delegator.Collection() == collectionID
+		return !delegatorFound
+	})
+	if delegatorFound {
+		return
+	}
+
+	segmentFound := len(node.manager.Segment.GetBy(segments.WithCollection(collectionID))) > 0
+	if segmentFound {
+		return
+	}
+
+	node.manager.Collection.Remove(collectionID)
```

Comment:
this logic check delegator first then do remove
while in `WatchDmlChannel` it puts a new collection then add it into delegators, which may cause this collection being released during concurrent processing